solution/index.js

```diff
@@ -0,0 +1,349 @@
+            'Content-Type': "application/json",
+        },
+        body: JSON.stringify({"tasks": localStorageObject})
+    })
+
+    hideLoading();
+
+    if(!response.ok){ // error part: behavior of all elements in error situation anf throw ERROR with current status
+        alert("ERROR "+response.status+". "+response.statusText)//displaying in result element at error message
+        throw `ERROR ${response.status}`
+    }
+    
+}
+
+////Fucntion of loading tasks from API Storage if in localStorage we have other tasks function refreshes localStorage information to API information 
+async function loadFromStorage(){
+    displayLoading();
+    const response =await fetch("https://json-bins.herokuapp.com/bin/614ee0bfc092dca86cb87689")
+    if(!response.ok){ // error part: behavior of all elements in error situation anf throw ERROR with current status
+        alert("ERROR "+response.status+". "+response.statusText)//displaying in result element at error message
+        throw `ERROR ${response.status}`
+    }
+    hideLoading();
+    const tasksOBjectFromStorage=(await response.json()).tasks
+    localStorageObject=tasksOBjectFromStorage;
+    localStorage.tasks=JSON.stringify(localStorageObject);
+    refreshTaskSection();
+    
+}
+
+//Starting drag event
+function dragEvent (event){
+    draggingElement = event.target.closest("li")
+    event.dataTransfer.setData("text", event.target.closest("ul").className);
+
+    const sections = document.querySelectorAll("section") //Adding event listener to all sections
+    for(let section of sections){
+        section.addEventListener("dragover", allowDrop)
+        section.addEventListener("drop", dropEvent)
+    }
+}
+
+//Preparing to drop event
+function allowDrop(event) {
+    event.preventDefault();
+}
+
+//Finish of dragging, drop event
+function dropEvent(event) {
+    event.preventDefault();
+
+    const previousParentClassName = event.dataTransfer.getData("text");
+    const newSectionClass = event.target.closest("section").className
+    const newParentList = event.target.closest("section").querySelector("ul")
+    const valueOfDraggingElement = draggingElement.textContent;
+
+    switch(newSectionClass){ //Check if I want to delete or replace task
+        case "dropzone":
+            newParentList.insertBefore(draggingElement, newParentList.childNodes[0]);
+            removeReplacingElement(valueOfDraggingElement);
+            let localStorageArray;
+            switch(newParentList.className){
+                case "in-progress-tasks":
+                    localStorageArray = localStorageObject["in-progress"]
+                    break;
+                case "to-do-tasks":
+                    localStorageArray = localStorageObject["todo"]
+                    break;
+                case "done-tasks":
+                    localStorageArray = localStorageObject["done"]
+                    break;
+            }
+            localStorageArray.unshift(valueOfDraggingElement)
```

Comment:
There is an unexpected behavior regarding the location of the element drop. For example when arranging elements within a section and try to change the position of the element to a higher index
it is initialized in the wrong place